src/main/flight/gps_rescue.c

```diff
@@ -567,6 +570,22 @@ static void sensorUpdate(void)
     rescueState.sensor.distanceToHomeM = rescueState.sensor.distanceToHomeCm / 100.0f;
     rescueState.sensor.groundSpeedCmS = gpsSol.groundSpeed; // cm/s
 
+    // values to modify dcm_kp to a low value when climbing, and a higher value when there is a velocity error
+    if (gpsRescueConfig()->rescueGroundspeed) {
+        const float setGroundspeedInv = 1.0f / gpsRescueConfig()->rescueGroundspeed;
+        float groundspeedErrorRatio = fabsf(rescueState.sensor.groundSpeedCmS - rescueState.sensor.velocityToHomeCmS) * setGroundspeedInv;
+        // zero if groundspeed = velocity to home,
+        // 1 if flying sideways at target groundspeed but zero home velocity,
+        // 2 if flying backwards at target groundspeed
```

Comment:
Probably more accurate, in a general sense, to say 'moving away from home'.  I'll do that.
You're right, the nose could be pointing towards home, or away from home (or any direction at all, but let's consider the two extremes).
If the nose is pointing towards home, pitching forward will work normally, accelerating towards to home, reducing the negative velocity to home.  
If the nose is pointing away from home, this can only be the case, after rotation is complete, if the IMU is 180 degrees wrong.  Pitching forward with twill just make the quad move faster away from home.  We need to correct the IMU error quickly, and to limit the pitch angle to minimise speed away from home.  The PR achieves that.

Reply:
(It is possible that IMU handles 180deg case very poorly). 
Also it would make sense to only increase GPS heading gain, attitude is probably fine.